game.sy

```diff
@@ -430,10 +430,43 @@ update :: fn delta: float -> void {
                 next_col = col
                 is_collision = true
             }
-
+            // Goal outside collision
+            arena.goals -> for_each' fn g: arena_mod.Goal {
+                col :: collision.circle_vs_circle' a, g.cage, delta * t * (a.vel - g.cage.vel)
+                if col.t <= 0.0 || next_col.t < col.t {
+                    ret
+                }
+                if abs(a.pos[0]) > abs(g.cage.pos[0]) || sign(col.normal[0]) == sign(g.cage.pos[0]) {
+                    ret
+                }
+                next_col = col
+                is_collision = true
+            }
+            // Goal inside collision
+            arena.goals -> for_each' fn g: arena_mod.Goal {
+                a.rad = -a.rad
+                col :: collision.circle_vs_circle' a, g.cage, delta * t * (a.vel - g.cage.vel)
+                a.rad = -a.rad
```

Comment:
:/
It doesn't matter which one you negate - it might be clearer if you negate the cage.